data/visium_chicken_heart/chicken_heart.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import os
+import re
+import shutil
+import scanpy as sc
+import squidpy as sq
+import anndata as an
+import pandas as pd
+import scipy
+import tarfile
+import git
+import json
+from urllib import request
+from PIL import Image, ImageSequence
+
+
+def data_retrieval(out):
+    """
+    general function with all steps
+    """
+    Image.MAX_IMAGE_PIXELS = 933120000
+    
+    path = os.path.join(out, 'tmp')
+    os.makedirs(path) 
+
+    print('loading data')
+    
+    # download hires images 
+    request.urlretrieve('https://figshare.com/ndownloader/files/42985297', os.path.join(path,'D4.tif'))
+    request.urlretrieve('https://figshare.com/ndownloader/files/42985300', os.path.join(path,'D7.tif'))
+    request.urlretrieve('https://figshare.com/ndownloader/files/42985303', os.path.join(path,'D10.tif'))
+    request.urlretrieve('https://figshare.com/ndownloader/files/42985294', os.path.join(path,'D14.tif'))
+    
+    #download and untar archive from GEO
+    
+    request.urlretrieve('https://ftp.ncbi.nlm.nih.gov/geo/series/GSE149nnn/GSE149457/suppl/GSE149457_RAW.tar', os.path.join(path,'GSE149457_RAW.tar'))
+    with tarfile.open(os.path.join(path,'GSE149457_RAW.tar')) as f:
+        f.extractall(path)
+        
+    #download metadata from github
+    
+    git.Git(path).clone('https://github.com/madhavmantri/chicken_heart.git')
+    
+    # temporal folders rearrangements
+    print('download is complete')
+    
+    samples = ['D4', 'D7', 'D10', 'D14']
+    for sample in samples:
+        os.makedirs(os.path.join(path,sample, "spatial")) 
+    
+    h5_files = [f for f in os.listdir(path) if re.match(r'.*spatial_RNAseq.*\.h5', f)]
+    for sample in samples:
+        name = next(obj for obj in h5_files if sample in obj)
+        os.rename(os.path.join(path,name), os.path.join(path, sample, 'filtered_feature_bc_matrix.h5'))
+
+    processed_path = os.path.join(path,'chicken_heart/data/chicken_heart_spatial_RNAseq_processed')
+    lowres_files = [f for f in os.listdir(processed_path) if re.match(r'.*\.png', f)]
+    for sample in samples:
+        name = next(obj for obj in lowres_files if sample in obj)
+        os.rename(os.path.join(processed_path,name), os.path.join(path, sample, 'spatial','tissue_lowres_image.png'))
+    
+    json_files = [f for f in os.listdir(processed_path) if re.match(r'.*\.json', f)]
+    for sample in samples:
+        name = next(obj for obj in json_files if sample in obj)
+        os.rename(os.path.join(processed_path,name), os.path.join(path, sample, 'spatial','scalefactors_json.json'))
+    
+    csv_files = [f for f in os.listdir(processed_path) if re.match(r'.*\.csv', f)]
+    for sample in samples:
+        name = next(obj for obj in csv_files if sample in obj)
+        os.rename(os.path.join(processed_path,name), os.path.join(path, sample, 'spatial', 'tissue_positions_list.csv'))
+        
+        
+    # transform into png
+
+    print('transforming tif into png, takes some time')
```

Comment:
Transformation from tif to png not necessary

Reply:
Otherwise it gives error 
`SyntaxError: not a PNG file`
`OSError: Could not find 'hires_image'`